streaming/src/main/scala/org/apache/spark/streaming/receiver/Receiver.scala

```diff
@@ -266,14 +266,16 @@ abstract class Receiver[T](val storageLevel: StorageLevel) extends Serializable
 
   /** Attach Network Receiver executor to this receiver. */
   private[streaming] def attachExecutor(exec: ReceiverSupervisor) {
-    assert(executor_ == null)
-    executor_ = exec
+    assert(_supervisor == null)
+    _supervisor = exec
   }
 
   /** Get the attached executor. */
-  private def executor: ReceiverSupervisor = {
-    assert(executor_ != null, "Executor has not been attached to this receiver")
-    executor_
+  private[streaming] def supervisor: ReceiverSupervisor = {
+    assert(_supervisor != null,
```

Comment:
@helena I think this gives better error message in the scenario that you hit with your custom receiver?